### PR TITLE
Respect SITK_FORBID_DOWNLOADS for Doxygen

### PR DESCRIPTION
**Pull request by dave3d**
When building the Documentation, it was downloading the file
InsightDoxygen.tag from the ITK web site, even when the
SITK_FORBID_DOWNLOADS cmake flag was on.  Now it will not
do the download.

### CODE REVIEW

**Comment by blowekamp** [comment]
Is there an option to specify a local Doxygen tag file?

**Comment by dave3d** [comment]
There isn't but I'll work on it.

File: Utilities/Doxygen/Doxygen.cmake [hunk 1]
```diff
@@ -14,18 +14,21 @@ if (BUILD_DOXYGEN)
...
+      add_custom_command( OUTPUT "${PROJECT_BINARY_DIR}/Documentation/Doxygen/InsightDoxygen.tag"
+        COMMAND ${CMAKE_COMMAND} -D "PROJECT_SOURCE_DIR:PATH=${PROJECT_SOURCE_DIR}"
+        -D "OUTPUT_PATH:PATH=${PROJECT_BINARY_DIR}/Documentation/Doxygen"
+        -P "${PROJECT_SOURCE_DIR}/Utilities/Doxygen/ITKDoxygenTags.cmake"
+        DEPENDS "${PROJECT_SOURCE_DIR}/Utilities/Doxygen/ITKDoxygenTags.cmake"
+        )
+      set(DOXYGEN_TAGFILES_PARAMETER "${PROJECT_BINARY_DIR}/Documentation/Doxygen/InsightDoxygen.tag=https://www.itk.org/Doxygen/html/")
+      set(ITK_DOXYGEN_TAGFILE CACHE PATH "${PROJECT_BINARY_DIR}/Utilities/Doxygen/InsightDoxygen.tag")
+    else()
+      set(ITK_DOXYGEN_TAGFILE CACHE PATH "" )
```
Comment (blowekamp) [inline Utilities/Doxygen/Doxygen.cmake]: umm... the syntax for `set` is:
 `set(<variable> <value> [[CACHE <type> <docstring> [FORCE]] | PARENT_SCOPE])`

https://cmake.org/cmake/help/v3.0/command/set.html

Is DOXYGEN_TAGFILES_PARAMETER needed?

File: Utilities/Doxygen/Doxygen.cmake [hunk 2]
```diff
@@ -14,18 +14,21 @@ if (BUILD_DOXYGEN)
...
-      )
-    set(DOXYGEN_TAGFILES_PARAMETER "${PROJECT_BINARY_DIR}/Documentation/Doxygen/InsightDoxygen.tag=https://www.itk.org/Doxygen/html/")
+    if (NOT SITK_FORBID_DOWNLOADS)
+      add_custom_command( OUTPUT "${PROJECT_BINARY_DIR}/Documentation/Doxygen/InsightDoxygen.tag"
+        COMMAND ${CMAKE_COMMAND} -D "PROJECT_SOURCE_DIR:PATH=${PROJECT_SOURCE_DIR}"
+        -D "OUTPUT_PATH:PATH=${PROJECT_BINARY_DIR}/Documentation/Doxygen"
+        -P "${PROJECT_SOURCE_DIR}/Utilities/Doxygen/ITKDoxygenTags.cmake"
+        DEPENDS "${PROJECT_SOURCE_DIR}/Utilities/Doxygen/ITKDoxygenTags.cmake"
+        )
+      set(DOXYGEN_TAGFILES_PARAMETER "${PROJECT_BINARY_DIR}/Documentation/Doxygen/InsightDoxygen.tag=https://www.itk.org/Doxygen/html/")
```
Comment (blowekamp) [inline Utilities/Doxygen/Doxygen.cmake]: "${ITK_DOXYGEN_TAGFILE}=${ITK_DOXYGEN_HTML_ROOT}"?

**Comment by blowekamp** [comment]
Can these 3 commits be squashed and a single commit message used?

**Pull request closed without merging**